tb/cocotb/soc/test_apb4_register_bank.py

```
@@ -0,0 +1,320 @@
+    T1  test_apb_setup_access_protocol  — APB SETUP→ACCESS handshake; pready high only in ACCESS
+    T2  test_partial_pstrb_write        — pstrb=0b0101 leaves bytes 1,3 unchanged
+    T3  test_rw_roundtrip               — write→read round-trip; pslverr always 0
+    T4  test_out_of_range               — OOR write dropped, OOR read returns 0, pslverr=0
+    T5  test_hw_wen_bypasses_wmask      — HW injection to RO reg6 bypasses WMASK
+    T6  test_wmask_zero_ro_register     — SW write to WMASK=0 reg is silently dropped
+    T7  test_sw_hw_collision_sw_wins    — same-cycle SW+HW write: SW takes priority
+"""
+
+import cocotb
+from cocotb.clock import Clock
+from cocotb.triggers import RisingEdge, Timer
```

Flake8 (7.3.0)

[error] 'cocotb.triggers.Timer' imported but unused

(F401)